### ISSUE TITLE
Standard Library Modules: error C2039: `'promise_type'`: is not a member of `'std::coroutine_traits<TestModule::FireAndForget>'`

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#4826]
Repros with VS 2022 17.11 Preview 3. Reported by CppMan on Discord.

I'm not familiar enough with coroutines, or with multiple module scenarios, to say what should happen when a module tries to export a partial specialization like `std::coroutine_traits<TestModule::FireAndForget, Ts...>`.

```
C:\Temp>type meow.ixx
```
```cpp
export module Module;

import std;

export namespace TestModule {
    struct FireAndForget {};
} // namespace TestModule

export template <typename... Ts>
struct std::coroutine_traits<TestModule::FireAndForget, Ts...> {
    struct promise_type {
        TestModule::FireAndForget get_return_object() const noexcept {
            return {};
        }

        void return_void() const noexcept {}

        std::suspend_never initial_suspend() const noexcept {
            return {};
        }

        std::suspend_never final_suspend() const noexcept {
            return {};
        }

        void unhandled_exception() const noexcept {}
    };
};
```
```
C:\Temp>type test.cpp
```
```cpp
import Module;
import std;

TestModule::FireAndForget Test() {
    co_return;
}

int main() {
    Test();
}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od /c "%VCToolsInstallDir%\modules\std.ixx"
std.ixx

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od /c meow.ixx
meow.ixx

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od test.cpp meow.obj std.obj
test.cpp
test.cpp(5): error C2039: 'promise_type': is not a member of 'std::coroutine_traits<TestModule::FireAndForget>'
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.41.34021\include\coroutine(46): note: see declaration of 'std::coroutine_traits<TestModule::FireAndForget>'
```

**Comment by cdacamar** [comment]
Yes, I can confirm this is a compiler bug.  There is a workaround:

meow.ixx:
```c++
export module Module;

import std;

export namespace TestModule {
    struct FireAndForget {};
} // namespace TestModule

namespace std { // Note: we remove the qualified name declaration.
export template <typename... Ts>
struct coroutine_traits<TestModule::FireAndForget, Ts...> {
    struct promise_type {
        TestModule::FireAndForget get_return_object() const noexcept {
            return {};
        }

        void return_void() const noexcept {}

        std::suspend_never initial_suspend() const noexcept {
            return {};
        }

        std::suspend_never final_suspend() const noexcept {
            return {};
        }

        void unhandled_exception() const noexcept {}
    };
};
}
```

**Comment by naddu77** [comment]
I am `CppMan`.
The same issue occurs with the `std::formatter` class template specialization.
The same can be solved by modifying as @cdacamar suggested.
Thank you.

**Comment by StephanTLavavej** [comment]
@cdacamar fixed this with internal MSVC-PR-565655 which will be available in VS 2022 17.12 Preview 2. Thanks again for the report!
